### DISCUSSION TITLE
Use CacheDataset for DDP

### DISCUSSION THREAD

**Discussion by hw-ju** [Project-MONAI/tutorials#672] [Q&A]
Hi! 
In brats_training_ddp.py, `BratsCacheDataset` is implemented to "_enhance the DecathlonDataset to support distributed data parallel_" and it will "_partition dataset based on current rank number, every rank trains with its own data. it can avoid duplicated caching content in each rank, but will not do global shuffle before every epoch_". 

Can the script still work if we replace the `BratsCacheDataset` object with a `CacheDataset` object and a `DistributedSampler` object, to get a global shuffle before every epoch? 

If it can work, then in this way, will the **whole** intermediate transformed dataset be cached on each GPU memory?

**Comment by Nic-Ma** [comment]
Hi @hw-ju ,

Yes, if you want to `cache` and `global shuffle`, every process will cache the whole data, because it can't shuffle across processes.

Thanks.

**Reply by hw-ju** [reply]
Thanks @Nic-Ma !
